community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/BigEndianByteArrayBuffer.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    static int compare( byte[] key, byte[] searchSpace, int offset )
+    {
+        for ( int i = 0; i < key.length; i++ )
+        {
+            int result = (key[i] & 0xFF) - (searchSpace[offset + i] & 0xFF);
+            if ( result != 0 )
+            {
+                return result;
+            }
+        }
+        return 0;
+    }
+
+    public void clear()
+    {
+        fill( (byte) 0 );
+    }
+
+    public void fill( byte zero )
+    {
+        Arrays.fill( buffer, zero );
+    }
+
+    public boolean allZeroes()
+    {
+        for ( byte b : buffer )
+        {
+            if ( b != 0 )
+            {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public void dataFrom( ByteBuffer buffer )
+    {
+        buffer.get( this.buffer );
+    }
+
+    public void dataTo( byte[] target, int targetPos )
+    {
+        assert target.length >= targetPos + buffer.length : "insufficient space";
+        System.arraycopy( buffer, 0, target, targetPos, buffer.length );
+    }
+
+    public int size()
+    {
+        return buffer.length;
+    }
+
+    public byte getByte( int offset )
+    {
+        if ( offset > size() - 1 )
+        {
+            throw new IndexOutOfBoundsException();
+        }
+        return buffer[offset];
+    }
+
+    public BigEndianByteArrayBuffer putByte( int offset, byte value )
+    {
+        return putValue( offset, value, 1 );
+    }
+
+    public short getShort( int offset )
+    {
+        if ( offset > size() - 2 )
+        {
+            throw new IndexOutOfBoundsException();
+        }
+        return (short) (((0xFF & buffer[offset]) << 8) |
+                        (0xFF & buffer[offset + 1]));
+    }
+
+    public BigEndianByteArrayBuffer putShort( int offset, short value )
+    {
+        return putValue( offset, value, 2 );
+    }
+
+    public char getChar( int offset )
+    {
+        if ( offset > size() - 2 )
+        {
+            throw new IndexOutOfBoundsException();
+        }
+        return (char) (((0xFF & buffer[offset]) << 8) |
+                       (0xFF & buffer[offset + 1]));
+    }
+
+    public BigEndianByteArrayBuffer putChar( int offset, char value )
+    {
+        return putValue( offset, value, 2 );
+    }
+
+    public int getInt( int offset )
+    {
+        if ( offset > size() - 4 )
+        {
+            throw new IndexOutOfBoundsException();
+        }
+        return (((0xFF & buffer[offset]) << 24) |
+                ((0xFF & buffer[offset + 1]) << 16) |
+                ((0xFF & buffer[offset + 2]) << 8) |
+                (0xFF & buffer[offset + 3]));
+    }
+
+    public BigEndianByteArrayBuffer putInt( int offset, int value )
+    {
+        return putValue( offset, value, 4 );
+    }
+
+    public long getLong( int offset )
+    {
+        if ( offset > size() - 8 )
+        {
+            throw new IndexOutOfBoundsException( String.format( "offset=%d, size=%d", offset, size() ) );
```

Comment:
Unify error messages in `IndexOutOfBoundsException`? Some exceptions have a message other not.